packages/splits-sdk/src/utils/index.ts

```diff
@@ -115,3 +132,99 @@ export const getRecoupTranchesAndSizes = async (
 
   return [recoupTranches, sizes]
 }
+
+export const getAddressAndAllocationFromRecipients = (
+  recipients: SplitRecipient[],
+): { recipientAddresses: Address[]; recipientAllocations: bigint[] } => {
+  return {
+    recipientAddresses: recipients.map(
+      (recipient) => recipient.address,
+    ) as Address[],
+    recipientAllocations: recipients.map((recipient) =>
+      getBigIntFromPercent(recipient.percentAllocation),
+    ),
+  }
+}
+
+export const MAX_DISTRIBUTION_INCENTIVE = 6.5535
+
+export const getValidatedSplitV2Config = (
+  recipients: SplitRecipient[],
+  distributorFeePercent: number,
+  totalAllocationPercent?: number,
+): {
+  recipientAddresses: Address[]
+  recipientAllocations: bigint[]
+  distributionIncentive: number
+  totalAllocation: bigint
+} => {
+  const { recipientAddresses, recipientAllocations } =
+    getAddressAndAllocationFromRecipients(recipients)
+
+  if (distributorFeePercent > MAX_DISTRIBUTION_INCENTIVE)
+    throw new InvalidDistributorFeePercentErrorV2(distributorFeePercent)
+  const distributionIncentive = getNumberFromPercent(distributorFeePercent)
+
+  const calculatedTotalAllocation = recipientAllocations.reduce((a, b) => a + b)
+
+  if (
+    totalAllocationPercent &&
+    getBigIntFromPercent(totalAllocationPercent) !== calculatedTotalAllocation
+  )
+    throw new InvalidTotalAllocation(totalAllocationPercent)
+  else if (calculatedTotalAllocation !== PERCENTAGE_SCALE)
+    throw new InvalidTotalAllocation()
+
+  return {
+    recipientAddresses,
+    recipientAllocations,
+    distributionIncentive,
+    totalAllocation: calculatedTotalAllocation,
+  }
+}
+
+export const getSplitType = (
+  chainId: number,
+  factoryAddress: Address,
+): SplitV2Type => {
+  if (factoryAddress === getSplitV2FactoryAddress(chainId, SplitV2Type.Pull))
+    return SplitV2Type.Pull
+  return SplitV2Type.Push
+}
+
+export const getAccountsAndPercentAllocations: (
```

Comment:
nit: getV1AccountsAndPercentAllocations? Or do we know if v2 is not in the name it's for v1?